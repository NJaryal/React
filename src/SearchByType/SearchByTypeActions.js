export const SET_SEARCH_BY_TITLE = 'SET_SEARCH_BY_TITLE';
export const SET_SEARCH_BY_GENRE = 'SET_SEARCH_BY_GENRE';

export const setSearchByTitle = () => ({ type: SET_SEARCH_BY_TITLE });
export const setSearchByGenre = () => ({ type: SET_SEARCH_BY_GENRE });
