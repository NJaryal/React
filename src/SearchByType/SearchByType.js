import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Button from 'Components/common/Button';
import { setSearchByTitle, setSearchByGenre } from './SearchByTypeActions';
import globalStyles from '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import styles from './SearchByType.css';

export const SearchByType = ({
  activeSearch, dispatchSetSearchByTitle, dispatchSetSearchByGenre,
}) => (
  <div className={globalStyles['btn-group']} role="group" aria-label="Basic example" data-test="activeSearch">
        SEARCH BY
    <Button className={`${activeSearch === 'title' ? styles.active : styles.inactive}`} onClick={dispatchSetSearchByTitle} children="Title" />
    <Button className={`${activeSearch === 'genre' ? styles.active : styles.inactive}`} onClick={dispatchSetSearchByGenre} children="Gengre" />
  </div>
);
SearchByType.propTypes = {
  activeSearch: PropTypes.string.isRequired,
  dispatchSetSearchByTitle: PropTypes.func.isRequired,
  dispatchSetSearchByGenre: PropTypes.func.isRequired,
};
const mapStateToProps = null;

const mapDispatchToProps = dispatch => ({
  dispatchSetSearchByTitle: () => {
    dispatch(setSearchByTitle());
  },
  dispatchSetSearchByGenre: () => {
    dispatch(setSearchByGenre());
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchByType);
