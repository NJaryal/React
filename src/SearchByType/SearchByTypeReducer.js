import { SET_SEARCH_BY_TITLE, SET_SEARCH_BY_GENRE } from './SearchByTypeActions';

function SearchByTypeReducer(state = {
  titleActive: true,
  genresActive: false,
}, action) {
  switch (action.type) {
    case SET_SEARCH_BY_TITLE:
      return {
        ...state,
        titleActive: true,
        genresActive: false,
      };
    case SET_SEARCH_BY_GENRE:
      return {
        ...state,
        titleActive: false,
        genresActive: true,
      };
    default:
      return state;
  }
}
export default SearchByTypeReducer;
