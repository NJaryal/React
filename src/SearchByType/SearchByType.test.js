import React from 'react';
import { shallow } from 'enzyme';
import { SearchByType } from './SearchByType';
import { findByTestAttr, checkProps } from '../../Utils';
import SearchByTypeReducer from './SearchByTypeReducer';
import { SET_SEARCH_BY_GENRE } from './SearchByTypeActions';

const renderSearchByType = (props = {}) => {
  const component = shallow(<SearchByType {...props} />);
  return component;
};

describe('SearchByType component', () => {
  describe('Check PropTypes', () => {
    it('should not throw a warning', () => {
      const expectedprops = {
        activeSearch: 'title',
      };
      const propdError = checkProps(SearchByType, expectedprops);
      expect(propdError).toBeDefined();
    });
  });

  describe('Have Props', () => {
    let wrapper;
    beforeEach(() => {
      const props = {
        activeSearch: 'title',
      };
      wrapper = renderSearchByType(props);
    });

    it('should render a activeSort', () => {
      const activeSort = findByTestAttr(wrapper, 'activeSearch');
      expect(activeSort.length).toBe(1);
    });
  });

  describe('reducer', () => {
    it('should return default state', () => {
      const newState = SearchByTypeReducer(undefined, {});
      expect(newState).toEqual({ 
        activeSearch: 'title',
      });
    });

    it('should return new state if receiving Type', () => {
      const SearchBy = { activeSearch: 'genre' };
      const newState = SearchByTypeReducer(undefined, {
        type: SET_SEARCH_BY_GENRE,
        payload: SearchBy,
      });
      expect(newState).toEqual(SearchBy);
    });
  });

  describe('should show', () => {
    let component;
    beforeEach(() => {
      component = renderSearchByType();
    });
    it('snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
