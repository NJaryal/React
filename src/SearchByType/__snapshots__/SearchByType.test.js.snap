// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchByType component should show snapshot 1`] = `
<div
  aria-label="Basic example"
  className="btn-group"
  data-test="activeSearch"
  role="group"
>
  SEARCH BY
  <Button
    className="inactive"
  >
    Title
  </Button>
  <Button
    className="inactive"
  >
    Gengre
  </Button>
</div>
`;
